cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/DatasetInstanceService.java

```diff
@@ -187,7 +198,8 @@ private DatasetMeta getFromMds(DatasetId instance) throws Exception {
     spec = DatasetsUtil.fixOriginalProperties(spec);
 
     DatasetTypeId datasetTypeId = instance.getParent().datasetType(spec.getType());
-    DatasetTypeMeta typeMeta = getTypeInfo(instance.getParent(), spec.getType());
+    // by pass the auth check for dataset type when the operation is not creation
+    DatasetTypeMeta typeMeta = getTypeInfo(instance.getParent(), spec.getType(), true);
```

Comment:
Let's add a comment to the javadoc of this method saying it should not be called during creation since it does not check authorization for dataset types.